Basic game screen design

SofiiaMal opened the pull request:
includes the basic html and css structure, a bit incteractivity with js

SofiiaMal committed:
Basic game screen design

includes the basic html and css structure, a bit incteractivity with js

ThePersonThat commented on js/scriptGame.js:
```diff
@@ -0,0 +1,130 @@
+let isInventoryOpen = false;
```
Let's have seperate file which contains inventory functionality, let's name it as "inventory.js"

ThePersonThat commented on js/scriptGame.js:
```diff
@@ -0,0 +1,130 @@
+let isInventoryOpen = false;
+
+document.getElementsByClassName("inventorySlot slotArrow")[0].onclick = function() {
```
What do you think about using .querySelector? https://www.w3schools.com/Jsref/met_document_queryselector.asp

ThePersonThat commented on js/scriptGame.js:
```diff
@@ -0,0 +1,130 @@
+let isInventoryOpen = false;
+
+document.getElementsByClassName("inventorySlot slotArrow")[0].onclick = function() {
+    if(!isInventoryOpen){
```
```suggestion
    if (!isInventoryOpen) {
```
Fix formating like this everywhere please

ThePersonThat commented on js/scriptGame.js:
```diff
@@ -0,0 +1,130 @@
+        isInventoryOpen = true;
+        document.getElementsByClassName("blackoutPanel")[0].classList.remove("hidden")
+    }
+    else if(isInventoryOpen){
```
```suggestion 
     else if (isInventoryOpen) {
```
Let's have if's like this
```
if () {
    // code
} else {
   // code
}
```

ThePersonThat commented on js/scriptGame.js:
```diff
@@ -0,0 +1,130 @@
+    else if(isInventoryOpen){
+        moveInventory("+", 40, 180, 0.5);
+        isInventoryOpen = false;
+        setTimeout(()=> {
```
```suggestion
        setTimeout(() => {
```

ThePersonThat commented on js/scriptGame.js:
```diff
@@ -0,0 +1,130 @@
+    triggerLevelPopUp(100, 0);
+    
+    setTimeout(()=> {
+        levelPopUp.classList.add("hidden");
```
Let's have separated function for hidden functionalitty. Change it everywhere where you opearte with "hidden" class. What dou you think about the following:

```suggestion
const HIDDEN_CLASS_NAME = "hidden";

function hideElement(elem) {
    elem.classList.add(HIDDEN_CLASS_NAME);
}

functoin unhideElement(elem) {
    elem.classList.remove(HIDDEN_CLASS_NAME);
}

function isElementHidden(elem) {
   return elem.classList.contains(HIDDEN_CLASS_NAME);
}
```

ThePersonThat commented on js/scriptGame.js:
```diff
@@ -0,0 +1,130 @@
+    
+    setTimeout(()=> {
+        levelPopUp.classList.add("hidden");
+        levelPopUp.classList.remove("victoryPop");
```
```suggestion
        levelPopUp.classList.remove("victoryPop", "levelPop", "defeatPop");
```

ThePersonThat commented on css/styleGame.css:
```diff
@@ -1,17 +1,329 @@
+    z-index: 2;
+    width: 100%;
+    height: 100%;
+   
```
Remove extra line

ThePersonThat commented on gameScreen.html:
```diff
@@ -8,6 +8,78 @@
     </head>
 
     <body class="">
```
Remove empty class

SofiiaMal committed:
Brief changes to structure and code

Two new js files :
- inventory.js for inventory functionality
- utility.js for small usefull things referenced in different files

Added .levelPop properties in styleGame.css, bc before it was added via js but didn't do anything visually

Added boss room icon to the assets

Minor changes to scriptGame.js :
- changed getElementsByClassName[0] to querrySelector
- merged several classList.remove into one
- inventory functionality moved to inventory.js
- fixed victory pop up text
- work with "hidden" class is now done via utility.js

ThePersonThat commented on js/inventory.js:
```diff
@@ -0,0 +1,42 @@
+    if (!isInventoryOpen) {
+        moveInventory("-", 85, 0, 0);
+        isInventoryOpen = true;
+        document.querySelector(".blackoutPanel").classList.remove("hidden");
```
Please, use utility.js here

ThePersonThat commented on js/inventory.js:
```diff
@@ -0,0 +1,42 @@
+    const duration = 300; // ms
+    const startTime = performance.now();
+    const endDeg =  direction === "+" ? startDeg + 180 : startDeg - 180;
+    const endOpacity = direction === "+" ? startOpacity - 0.5 : startOpacity + 0.5;
```
What do u think about this order?
```suggestion
    const element = document.querySelector(".playerInventory");
    
    const duration = 300; // ms
    const startTime = performance.now();
    
    const endVh = direction === "+" ? startVh + 45 : startVh - 45;
    const endDeg =  direction === "+" ? startDeg + 180 : startDeg - 180;
    const endOpacity = direction === "+" ? startOpacity - 0.5 : startOpacity + 0.5;
```

ThePersonThat commented on js/inventory.js:
```diff
@@ -0,0 +1,42 @@
+        const progress = Math.min(elapsed / duration, 1); // clamp between 0 and 1
+
+        // Interpolate between start and end
+        const currentVh = startVh + (endVh - startVh) * progress;
```
What do u think about this order?
```suggestion
        const currentVh = startVh + (endVh - startVh) * progress;
         const currentDeg = startDeg + (endDeg - startDeg) * progress;
        const currentOpacity = startOpacity + (endOpacity - startOpacity) * progress;
        
        element.style.top = `${currentVh}vh`;
        document.querySelector(".inventorySlot.slotArrow").style.rotate = `${currentDeg}deg`
        document.querySelector(".blackoutPanel").style.opacity = currentOpacity;
```

SofiiaMal committed:
minor fixes

formating in css, an unused element in html, use of utility.js in inventory.js and minor order changes

ThePersonThat reviewed (approved):
Don't forget to squash :)

ThePersonThat commented on js/scriptGame.js:
```diff
@@ -0,0 +1,79 @@
+
+//temporal bit till we get the game sequense going
+
+document.getElementsByClassName("controlButton")[0].onclick = function() {
```
I did not review this part as it's goingo to be rewritten

VolodymyrVH reviewed (approved):
Everything is fine

ThePersonThat commented on js/inventory.js:
```diff
@@ -0,0 +1,45 @@
+    } else if (isInventoryOpen) {
+        moveInventory("+", 40, 180, 0.5);
+        isInventoryOpen = false;
+        setTimeout(() => {
```
Minor: But fix it in the next your PRs
```suggestion
const panel = document.querySelector(".blackoutPanel");

if (!isInventoryOpen) {
  ....
   utilityScript.unhideElement(panel);
} else { // remove if check
   ....
   setTimeout(() => utilityScript.hideElement(panel), 400);
}
